vllm/model_executor/models/llama.py

```diff
@@ -427,7 +427,7 @@ def __init__(
                 quant_config=quant_config,
             )
             if config.tie_word_embeddings:
-                self.lm_head.weight = self.model.embed_tokens.weight
+                self.lm_head = self.model.embed_tokens
```

Comment:
@Isotr0py Can you justify this change/ has it been tested? Seems like a bug to me, as lm_head and embed_tokens are different types and serve different functions

Reply:
Will this be a problem on other models which only assign weights and not modules?

Reply:
I understand that this may be out of scope for these changes

Reply:
For those [models support GGUF](https://huggingface.co/docs/transformers/gguf#supported-model-architectures), corresponding to `LlamaForCausalLM` and `Qwen2ForCausalLM` in vllm (Qwen2moe hasn't supported GGUF in vllm yet), this might be not a problem. 

Because `Qwen2` has assigned modules in their implementation:
https://github.com/vllm-project/vllm/blob/01843c89b8ddae00d4a0f0f56b8aa7fbaa3efc42/vllm/model_executor/models/qwen2.py#L351-L356

Reply:
Not sure if it's just me but I'm unable to start vllm with `--model=meta-llama/Llama-3.2-1B --enable-lora` now: 
```
RuntimeError: The size of tensor a (2048) must match the size of tensor b (128512) at non-singleton dimension 
```